lib/src/json_schema/json_schema.dart

```diff
@@ -1305,14 +1304,15 @@ This functionality will be removed in 3.0.
           template += '#${_ref.fragment}';
         }
         _ref = Uri.parse(template);
-
         // If the $id has a fragment, append it to the base, or use it alone.
       } else if (_ref.fragment != null && _ref.fragment.isNotEmpty) {
         _ref = Uri.parse('${_inheritedUri ?? ''}#${_ref.fragment}');
       }
     }
 
-    if (_ref.scheme.isNotEmpty) {
+    // The ref's base is a relative file path, so it should be treated as a relative file URI
+    final isRelativeFileUri = _inheritedUriBase != null && _inheritedUriBase.scheme.isEmpty;
```

Comment:
should we check that we're in dart:io or something here too? Seems like we wouldn't want to somehow fall into this trap for HTTP(S) resolution

Reply:
From hat I can tell, this should be harmless for browser usage — if the base URI scheme is empty it will fail on attempts to fetch the file.

I tried the test case in the browser suite, and got this error whether or not the above logic change was in place:

```
RequestException: GET 404 Not Found test/relative_refs/root.json
```